subsys/net/ip/tcp2.c

```diff
@@ -208,6 +208,13 @@ static void tcp_send_queue_flush(struct tcp *conn)
 				struct net_pkt, next))) {
 		tcp_pkt_unref(pkt);
 	}
+
+	(void)k_delayed_work_cancel(&conn->data_timer);
+
+	if (conn->pending_data) {
+		tcp_pkt_unref(conn->pending_data);
+		conn->pending_data = NULL;
+	}
```

Comment:
Shall we just move it into tcp_conn_unref()?

Reply:
Can be done. Not sure why we even need tcp_send_queue_flush() as it is only called from tcp_conn_unref()